### PR TITLE
nixos/manticore: init module

### PR DESCRIPTION
**Pull request by onny**
###### Description of changes

Simple module for the open-source Elasticsearch alternative [manticore](https://manticoresearch.com/)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/search/manticoresearch/default.nix [hunk 1]
```diff
@@ -60,6 +60,11 @@ stdenv.mkDerivation rec {
     "-DWITH_GALERA=0"
   ];
 
+  postFixup = ''
+    # Remove references to nix store in dist config file
+    sed -i 's|/nix/store/.*/var|/var|' $out/etc/manticoresearch/manticore.conf
+  '';
+
```
Comment (jdelStrother) [inline pkgs/servers/search/manticoresearch/default.nix:67]: ah, thanks - I wondered what a good way of setting sane paths in the default config was.

**Comment by jdelStrother** [comment]
I'm not sure I'm qualified to review this because I don't generally use NixOS, but from the POV of a manticore user it looks good.

File: pkgs/servers/search/manticoresearch/default.nix [hunk 2]
```diff
@@ -60,6 +60,11 @@ stdenv.mkDerivation rec {
     "-DWITH_GALERA=0"
   ];
 
+  postFixup = ''
+    # Remove references to nix store in dist config file
+    sed -i 's|/nix/store/.*/var|/var|' $out/etc/manticoresearch/manticore.conf
```
Comment (SuperSandro2000) [inline pkgs/servers/search/manticoresearch/default.nix:65]: We can't fix this easily with a configure flag or similar?

File: nixos/modules/services/search/manticore.nix [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+          ProtectHome = true;
+          ProtectKernelLogs = true;
+          ProtectKernelModules = true;
+          ProtectKernelTunables = true;
+          SystemCallArchitectures = "native";
+          SystemCallFilter = [ "@system-service" "~@privileged" ];
+          RestrictRealtime = true;
+          LockPersonality = true;
+          MemoryDenyWriteExecute = true;
+          UMask = "0066";
```
Comment (SuperSandro2000) [inline nixos/modules/services/search/manticore.nix:68]: global read?

File: nixos/modules/services/search/manticore.nix [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+          ExecStart = "${pkgs.manticoresearch}/bin/searchd --config ${cfg.configFile}";
+          ExecStop = "${pkgs.manticoresearch}/bin/searchd --config ${cfg.configFile} --stopwait";
+          KillMode = "process";
+          KillSignal = "SIGTERM";
+          SendSIGKILL = "no";
+          LimitNOFILE = 65536;
+          LimitCORE = "infinity";
+          LimitMEMLOCK = "infinity";
+          Restart = "on-failure";
+          ExecStartPre="${pkgs.coreutils}/bin/mkdir -p /var/run/manticore";
```
Comment (SuperSandro2000) [inline nixos/modules/services/search/manticore.nix:44]:
```suggestion
```
Use tmpfiles.d

File: nixos/modules/services/search/manticore.nix [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+  config = mkIf cfg.enable {
+
+    systemd = {
+      services.manticore = {
+        wantedBy = [ "multi-user.target" ];
+        description = "Manticore Search full-text search solution";
+        after = [ "network.target" ];
+        serviceConfig = {
+          # Took config from upstream service https://github.com/manticoresoftware/manticoresearch/blob/master/dist/deb/manticore.service.in
```
Comment (SuperSandro2000) [inline nixos/modules/services/search/manticore.nix:34]: Please use the upstream systemd file and add our changes as a dropin which should be done automatically.

File: nixos/modules/services/search/manticore.nix [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+  options = {
+    services.manticore = {
+
+      enable = mkEnableOption (mdDoc "Manticore Search full-text search solution");
+
+      configFile = mkOption {
+        type = types.path;
+        default = "${pkgs.manticoresearch}/etc/manticoresearch/manticore.conf";
+        defaultText = literalExpression ''"''${pkgs.manticoresearch}/etc/manticoresearch/manticore.conf"'';
```
Comment (SuperSandro2000) [inline nixos/modules/services/search/manticore.nix:19]:
```suggestion
        defaultText = literalExpression "''${pkgs.manticoresearch}/etc/manticoresearch/manticore.conf";
```

File: nixos/modules/services/search/manticore.nix [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+  cfg = config.services.manticore;
+
+in {
+
+  options = {
+    services.manticore = {
+
+      enable = mkEnableOption (mdDoc "Manticore Search full-text search solution");
+
+      configFile = mkOption {
```
Comment (SuperSandro2000) [inline nixos/modules/services/search/manticore.nix:16]: This should be a settings type if possible.

**Comment by happysalada** [comment]
you've got a docbook ci failure https://github.com/NixOS/nixpkgs/actions/runs/3395260156/jobs/5644904171

Appart from that it looks good, did you leave it in draft to test more ?

**Comment by onny** [comment]
I fixed generating the custom "Sphinx search"-configuration-file format. Seems to work fine now for me :)

**Pull request merged**